src/mon/Monitor.cc

```diff
@@ -3010,9 +3017,17 @@ void Monitor::get_cluster_status(stringstream &ss, Formatter *f)
 	mgrmon()->get_map().print_summary(nullptr, &ss);
 	ss << "\n";
       }
-      if (mdsmon()->should_print_status()) {
-        ss << "    mds: " << spacing << mdsmon()->get_fsmap() << "\n";
+
+      const FSMap *fsmapp = &mdsmon()->get_fsmap();
+      if (!fsids.empty()) {
+	FSMap map = *fsmapp;
+	map.filter(fsids);
+	fsmapp = &map;
```

Comment:
ditto

Reply:
Fixed it. Let me know if it looks good - https://github.com/ceph/ceph/pull/32581/files#diff-d4e1ba36dd08617ea271ab2be3dbcb5bR3022

Reply:
LGTM